RS-Project10/Screen.java

```diff
@@ -0,0 +1,234 @@
+                    System.out.println("Press 3 Sort using timestamp");
+                    System.out.println("Press 4 Sort using distance from origin");
+                    int sortChoice = sc.nextInt();
+                    if (sortChoice == 1) {
+                        operation.sortUsingArea(listOfShapes);
+                    } else if (sortChoice == 2) {
+                        operation.sortUsingPerimeter(listOfShapes);
+                    } else if (sortChoice == 3) {
+                        operation.sortUsingTimestamp(listOfShapes);
+                    } else if (sortChoice == 4) {
+                        operation.sortUsingOriginDistance(listOfShapes);
+                    } else {
+                        System.out.println("Invalis operation");
+                        System.exit(0);
+                    }
+                    break;
+                case 6:
+                    int xCoordinate,
+                    yCoordinate;
+                    System.out.println("Enter x coordinate of point");
+                    xCoordinate = sc.nextInt();
+                    System.out.println("Enter y coordinate of point");
+                    yCoordinate = sc.nextInt();
+                    if (!operation.checkCoordinates(xCoordinate, yCoordinate)) {
+                        System.out.println("coordinates are wrong");
+                        System.exit(0);
+                    }
+                    Point point = new Point(xCoordinate, yCoordinate);
+                    ArrayList<Shape> listOfShapesEnclosingPoint;
+                    listOfShapesEnclosingPoint = operation.getShapesEnclosingPoint(listOfShapes, point);
+                    operation.displayShapes(listOfShapesEnclosingPoint);
+                    break;
+                case 7:
+                    operation.displayShapes(listOfShapes);
+                    System.out.println("Enter index of shape");
+                    int indexOfList = sc.nextInt();
+                    ArrayList<Shape> listOfShapesAbove;
+                    listOfShapesAbove = operation.getAboveShapes(listOfShapes, indexOfList - 1);
+                    operation.displayShapes(listOfShapesAbove);
+                    break;
+                case 8:
+                    operation.displayShapes(listOfShapes);
+                    System.out.println("Enter index of shape");
+                    int indexList = sc.nextInt();
+                    Point origin = listOfShapes.get((indexList-1)).getOrigin();
+                    System.out.println("X coordinate is " + origin.getXCoordinate());
+                    System.out.println("Y coordinate is " + origin.getYCoordinate());
+                    break;
+                default :
+                    System.out.println("Invalid choice");
+                    break;
+                }
+            } catch(Exception e) {
+                System.out.println("Enter valid inputs");

```

Comment:
if any other error will come, it will still say "enter valid input" - please get the error message from the exception object and check if error is related to invalid input.